cpp/src/groupby/hash/groupby.cu

```diff
@@ -166,10 +168,20 @@ void sparse_to_dense_results(
     [&col, to_dense_agg_result, mr, stream]
     (auto const& agg_kind) {
       auto tranformed_agg = std::make_unique<aggregation>(agg_kind);
-      auto argmax_result = to_dense_agg_result(tranformed_agg);
-      auto transformed_result = experimental::detail::gather(
-        table_view({col}), *argmax_result, false, false, false, mr, stream);
-      return std::move(transformed_result->release()[0]);
+      auto arg_result = to_dense_agg_result(tranformed_agg);
+      if (arg_result->has_nulls()) {
+        auto replacement = numeric_scalar<size_type>(-1, true, stream);
+        auto null_replaced_map = cudf::detail::replace_nulls(
```

Comment:
Yikes, groupbys on strings are going to be very expensive. 

The argmin/argmax results are already initialized to `ARGMIN/MAX_SENTINEL`. Won't null values already be initialized to the sentinel value?

Reply:
So the other option i can think of is this: I would keep track of whether the argmin/max request came as a result of a string min/max request. If so, the allocation of sparse results would not contain a null mask. Then, they'll be all valid and the untouched values would be sentinel. 

Need to figure out how to make this round trip through the result cache as it cannot distinguish between argmin with nulls and argmin without nulls. Meaning if we had an original argmin request along with a string min, the values would clash in the cache's map.

It would me much more convenient if there was a mechanism to create a column_view that simply drops the null mask. Because that's all I need.

Reply:
> It would me much more convenient if there was a mechanism to create a column_view that simply drops the null mask. Because that's all I need.

You could do `set_null_mask(rmm::device_buffer{})` to drop the null mask from a column.

Reply:
To your other points, perhaps the entries in the `result_cache` need some kind of reference to their "root/parent" aggregation request that generated them.

Reply:
I don't want to drop it from the owning column. That would mess with the original argmin request if any.

Reply:
I've made the changes but they look fragile. Changing `ARGMAX_SENTINEL` would break it.

Reply:
So what's the plan with this PR then?

Reply:
It’s just a choice between the two approaches. Drop null mask or fill nulls? After that I think it can be merged.

Reply:
Please make a choice and document it. If this is WIP please remove "review" so it doesn't get merged since it already has two approvals.

Reply:
Went with Jake's suggestion. Not WIP now.